src/mesh/mod.rs

```diff
@@ -213,35 +226,72 @@ impl ShapeMesh {
             }
         }
 
-
         Ok(())
     }
 
-    /// Returns the point on the shape's surface the camera is currently looking at
-    pub fn get_focus(&self, camera: &Camera) -> Option<Point3<f32>> {
+    /// Returns equally distributed points on the near_plane. The number of points
+    /// returned is focus_points².
+    pub fn get_focii(&self, camera: &Camera, focus_points: u8) -> Vec<Point3<f32>> {
+        // sets the amount of focus points that may cause a split event
         const EPSILON: f32 = 0.000_001;
         const MAX_ITERS: u64 = 100;
 
-        let mut pos = camera.position;
-        for _ in 0..MAX_ITERS {
-            let distance = self.shape.min_distance_from(pos);
-            pos += camera.direction() * distance;
-            if distance < EPSILON {
-                break;
+        let bb = camera.get_near_plane_bb();
+        let frustum_width = bb.1.x - bb.0.x;
+        let frustum_height = bb.1.y - bb.0.y;
+        let size_horizontal = frustum_width/focus_points as f32;
+        let size_vertical = frustum_height/focus_points as f32;
+        let center_diff = (bb.1 - bb.0)/(2.0*focus_points as f32);
+
+        let inv_view_trans = camera.view_transform().invert().unwrap();
```

Comment:
I think we want a `inv_view_transform()` method for `Camera`. That is mainly because we have an `unwrap()` here which I always like to explain in a comment. But explaining why the `unwrap()` is fine (explaining why the view matrix can always be inverted) every time we need the inverted matrix is stupid. So yes, I think we want said method :wink: